### PR TITLE
Rustup

### PR DESCRIPTION
**Pull request by bjorn3**
Will update when rust-lang/rust#55179 has landed.

I switched to the 2018 edition for `try` blocks.

### CODE REVIEW

File: src/render/locals.rs [hunk 1]
```diff
@@ -25,41 +27,47 @@ pub fn render_locals<'a, 'tcx: 'a>(
...
+                })
+            } else {
+                match *locals.get(id).unwrap() /* never None, because locals has a entry for every defined local */ {
+                    LocalValue::Dead => None,
+                    LocalValue::Live(op) => {
+                        if ecx.frame() as *const _ == frame as *const _ {
+                            Some(ecx.eval_operand(&mir::Operand::Move(mir::Place::Local(id)), None).unwrap())
+
+                        } else {
+                            None // TODO Above doesn't work for non top frames
```
Comment (bjorn3) [inline src/render/locals.rs:52]: TODO

File: src/render/mod.rs [hunk 2]
```diff
@@ -182,6 +183,8 @@ pub fn render_main_window(
     )
 }
 
+// TODO Memory::allocations doesn't exist anymore
```
Comment (bjorn3) [inline src/render/mod.rs:186]: Remainder to self

Comment (oli-obk) [reply src/render/mod.rs:186]: oops. yea, that should be trivial to reintroduce. Do you want to add it to your rustc PR? (Again, with a comment that priroda needs this)

~~**alternatively** (or just do it) you can do something super evil:~~

~~Implement `AllocMap` for a new ZST type that you create. Forward every method to a `FxHashMap` that you created in a lazy static. Now you have full access to the underlying memory by accessing the lazy static ;)~~

Ok, don't do that. A simple getter function should be ok to have in rustc. Remember to add a comment about priroda

Comment (bjorn3) [reply src/render/mod.rs:186]: Added to my rustc PR

**Comment by oli-obk** [comment]
I'll merge and keep working on top of it

**Comment by bjorn3** [comment]
I forgot about this PR :(

**Comment by oli-obk** [comment]
The master branch is back in business. Not everything works though (you commented out a few things that I didn't look at yet)

**Pull request merged**